src/hooks/useHarvest.ts

```diff
@@ -12,7 +12,7 @@ export const useHarvest = (farmPid: number) => {
 
   const handleHarvest = useCallback(async () => {
     const txHash = await harvest(masterChefContract, farmPid, account)
-    dispatch(fetchFarmUserDataAsync(account))
+    dispatch(fetchFarmUserDataAsync({ account, pids: [farmPid] }))
```

Comment:
This handler now only updates the relevant farm.

Reply:
Should we move it to the consumer side as with other hooks?